File: src/JFrameDemo2.java
import java.awt.Dimension;
import java.awt.GridLayout;
import java.awt.Toolkit;
import javax.swing.*;
import javax.swing.border.Border;

public class JFrameDemo2 {
    private static void constructGUI(){
        JFrame.setDefaultLookAndFeelDecorated(true);
        //建立JFrame
        JFrame frame = new JFrame();
        //建立空白邊框的contentPanel
        JPanel contentPanel = new JPanel();
        Border padding = BorderFactory.createEmptyBorder(10, 10, 10, 10);
        contentPanel.setBorder(padding);
        frame.setContentPane(contentPanel);

        frame.setDefaultCloseOperation(JFrame.EXIT_ON_CLOSE);
        frame.setTitle("JFrame Test");
        //建立frame的Layout

        frame.setLayout(new GridLayout(3,2,0,10));

        JLabel firstNameLabel = new JLabel("First Name:");
        firstNameLabel.setHorizontalAlignment(JLabel.RIGHT);
        frame.add(firstNameLabel);
        frame.add(new JTextField());

        JLabel lastNameLabel = new JLabel("Last Name:");
        lastNameLabel.setHorizontalAlignment(JLabel.RIGHT);
        frame.add(lastNameLabel);

        frame.add(new JTextField());

        frame.add(new JButton("Register"));

        int frameWidth = 300;
        int frameHeight = 150;
        Dimension screenSize = Toolkit.getDefaultToolkit().getScreenSize();
        frame.setBounds((int)screenSize.getWidth() - frameWidth, 0 , frameWidth, frameHeight);
        frame.setVisible(true);

    }

    public static void main(String[] args){
        /*
        SwingUtilities.invokeLater(new Runnable() {
            @Override
            public void run() {
                constructGUI();
            }
        });
        */
         SwingUtilities.invokeLater(() -> constructGUI());
    }
}

File: src/JFrameDemo2.kt
package FrameDemo1
import java.awt.GridLayout
import java.awt.Toolkit
import javax.swing.*

fun constructGUI(){
    JFrame.setDefaultLookAndFeelDecorated(true)
    val frame = JFrame()
    //建立空白邊框的contentPanel
    val contentPanel = JPanel()
    val padding = BorderFactory.createEmptyBorder(10, 10, 10, 10)
    contentPanel.border = padding
    //視窗關閉,JVM同時關閉
    frame.defaultCloseOperation = JFrame.EXIT_ON_CLOSE
    frame.title = "JFrame Test"
    frame.layout = GridLayout(3, 2, 0, 10)

    //建立FirstNameLabel
    val firstNameLabel = JLabel("First Name:")
    firstNameLabel.horizontalAlignment = JLabel.RIGHT
    frame.add(firstNameLabel)

    frame.add(JTextField())

    //建立LastNameLabel
    val lastNameLabel = JLabel("Last Name")
    lastNameLabel.horizontalAlignment = JLabel.RIGHT
    frame.add(lastNameLabel)

    frame.add(JTextField())

    frame.add(JButton("Register"))

    val frameWidth = 300
    val frameHeight = 150
    val screenSize = Toolkit.getDefaultToolkit().screenSize
    frame.setBounds(screenSize.width - (frameWidth+30), 0, frameWidth, frameHeight )
    frame.isVisible = true

}
fun main(){
    SwingUtilities.invokeLater{
        constructGUI()
    }
}